x/gamm/keeper/math.go

```diff
@@ -54,6 +54,21 @@ func calcSpotPriceWithSwapFee(
 	return spotPrice.Mul(scale)
 }
 
+func calcTokenGivenToken(
```

Comment:
Mentioned in DM, I think this is great. Should we rename the fn to `cfmmGetTokenOut` or something?

What we’re doing is basically saying “we are in a CFMM, that means f(balance A, balanceB) = f(balance A', balance B') . So I can always find the out amount by giving Balance A, Balance B, Balance A' (or equivalently with B, B’) because its symmetric 
(This assumes the CFMM invariant is symmetric in its dependency on A/B, which it is for Balancer, Stableswap, etc)

Reply:
Just renamed it to `constantFunctionInvariantSolver`